### ISSUE TITLE
False positive with redundant_clone lint

### ISSUE THREAD

**Issue by Heychsea** [rust-lang/rust-clippy#5690]
Hello, after updating to Rust stable 1.44.0 (and clippy with it) via rustup, I have encountered a false positive with the **redundant_clone** lint. 

```rust
fn foo(_: threadpool::ThreadPool) {
  // Do something with the pool
}

fn main() {
    let pool = threadpool::ThreadPool::new(4);
    pool.clone().execute(move || {
        foo(pool)
    })
}
```

Here is a link to the [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=886e9f8a3bdda00c1b0f2f6474752e5e), I don't really know how to explain what is happening but I know that it is not possible to remove the clone in my example as I am moving the value while borrowing it.

Thanks!

Here is my clippy version:
```shell
❯ cargo clippy -V                                                         
clippy 0.0.212 (d4092ace 2020-05-11)
```

**Comment by flip1995** [comment]
Duplicate of #5595
